### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by thegoldenmule**
- Moved some Async and Http extensions to those projects.
- Removed usage of MutableAsyncToken. <-- This is the most intense change. It _appears_ that scene loading thrashes around because of the tricky async nature of content loading.
- Deleted tons of unused classes.

### CODE REVIEW

File: Assets/Editor/Tools/Application/EditorBootstrapper.cs [hunk 1]
```diff
@@ -26,6 +28,25 @@ public void BootstrapCoroutine(IEnumerator coroutine)
             _coroutines.Add(coroutine);
         }
 
+        /// <inheritdoc />
+        public void Wait(float milliseconds, Action callback)
```
Comment (thegoldenmule) [inline Assets/Editor/Tools/Application/EditorBootstrapper.cs]: This was added to `IBootstrapper` which is part of the Http library.

File: Assets/Source/Async/Async.cs [hunk 2]
```diff
@@ -32,172 +32,6 @@ public TimedOutException() : base("Operation timed out.")
             }
         }
 
-        /// <summary>
```
Comment (thegoldenmule) [inline Assets/Source/Async/Async.cs]: Now in the Async library.

File: Assets/Source/Player/Element/Touch/TouchManager.cs [hunk 3]
```diff
@@ -360,7 +360,13 @@ private void TeardownElement(Element element)
         /// <param name="content">The content.</param>
         private void SetupContent(ContentWidget content)
         {
-            content.OnLoaded.OnSuccess(Content_OnAssetLoaded);
+            content.OnContentLoaded -= Content_OnAssetLoaded;
+            content.OnContentLoaded += Content_OnAssetLoaded;
+
+            if (content.IsContentLoaded)
```
Comment (thegoldenmule) [inline Assets/Source/Player/Element/Touch/TouchManager.cs]: This is really the only place we were counting on that mutable async token.

File: Assets/Source/Player/IUX/Content/ContentWidget.cs [hunk 4]
```diff
@@ -514,12 +474,21 @@ private void UpdateAsset()
...
             }
-
+            
             LogVerbose("Refresh asset for {0}.", Id);
 
             var assetId = _srcAssetProp.Value;
             var version = -1;
 
+            // there may not be a valid asset id (36 characters for a GUID + optional version)
+            if (assetId.Length < 36)
```
Comment (thegoldenmule) [inline Assets/Source/Player/IUX/Content/ContentWidget.cs]: We were doing TONS of stuff (creating tokens and arrays just to immediately resolve them) even if an asset was not assigned to an element.

File: Assets/Source/Player/IUX/Content/ContentWidget.cs [hunk 5]
```diff
@@ -545,6 +514,10 @@ private void UpdateAsset()
         private string[] GetScriptIds()
         {
             var scriptsSrc = _scriptsProp.Value;
+            if (scriptsSrc.Length <= 2)
```
Comment (thegoldenmule) [inline Assets/Source/Player/IUX/Content/ContentWidget.cs]: Early out for scripts not being assigned. The 2 comes from the fact that for now, if you added then removed a script, the prop value will be `[]`.

File: Assets/Source/Player/IUX/Content/ContentWidget.cs (hunk 4 above)
Comment (michaelcvoigt) [reply Assets/Source/Player/IUX/Content/ContentWidget.cs]: @thegoldenmule - Do we use Yoda conditions for these type of IF statements also?

File: Assets/Source/Player/IUX/Content/ContentWidget.cs (hunk 5 above)
Comment (jsundra) [reply Assets/Source/Player/IUX/Content/ContentWidget.cs]: Can this comment be added as a comment in code? I know I'll scratch my head a month from now when I look at this :)

File: Assets/Source/Player/IUX/Content/ContentWidget.cs [hunk 6]
```diff
@@ -667,9 +639,10 @@ private void PollScriptLoads()
 
             LogVerbose("All scripts loaded. Running!");
 
-            _onScriptsLoaded.Succeed(this);
-
             _runner.Setup(_enkluScripts);
+
+            IsContentLoaded = true;
```
Comment (jsundra) [inline Assets/Source/Player/IUX/Content/ContentWidget.cs]: Should `IsContentLoaded` be set back to false when `UpdateAsset` is called again?

Comment (thegoldenmule) [reply Assets/Source/Player/IUX/Content/ContentWidget.cs]: :+1:

File: Assets/Source/Player/IUX/Content/ContentWidget.cs (hunk 5 above)
Comment (thegoldenmule) [reply Assets/Source/Player/IUX/Content/ContentWidget.cs]: :+1:

File: Assets/Source/Player/IUX/Content/ContentWidget.cs (hunk 4 above)
Comment (thegoldenmule) [reply Assets/Source/Player/IUX/Content/ContentWidget.cs]: Just for `==`. And I don't think you _have_ to use yogish.

**Pull request merged**